scripts/performance/memory-performance/compute-memory-performance.js

```diff
@@ -0,0 +1,111 @@
+const TASK_DURATION = 1000
+const ACTION_NAMES = [
+  'RUM - add global context',
+  'RUM - add action',
+  'RUM - add error',
+  'RUM - add timing',
+  'RUM - start view',
+  'RUM - start/stop session replay recording',
+  'Logs - log message',
+]
+
+async function computeMemoryPerformance() {
+  const results = []
+  const pr = await fetchPR(LOCAL_BRANCH)
+  const bundleUrl = pr
+    ? `https://www.datad0g-browser-agent.com/pull-request/${pr.number}/datadog-rum.js`
+    : 'https://www.datadoghq-browser-agent.com/datadog-rum-canary.js'
+  const benchmarkUrl = pr
+    ? `https://datadoghq.dev/browser-sdk-test-playground/performance/?prNumber=${pr.number}`
+    : 'https://datadoghq.dev/browser-sdk-test-playground/performance/'
+  for (let i = 0; i < ACTION_NAMES.length; i++) {
+    const sdkTask = ACTION_NAMES[i]
+    const allBytesMeasurements = []
+    const allPercentageMeasurements = []
+    for (let j = 0; j < NUMBER_OF_RUNS; j++) {
+      const { medianPercentage, medianBytes } = await runTest(i, sdkTask, bundleUrl, benchmarkUrl)
+      allPercentageMeasurements.push(medianPercentage)
+      allBytesMeasurements.push(medianBytes)
+    }
+    const sdkMemoryPercentage = Number(
+      (allPercentageMeasurements.reduce((a, b) => a + b, 0) / allPercentageMeasurements.length).toFixed(2)
+    )
+    const sdkMemoryBytes = Number(
+      (allBytesMeasurements.reduce((a, b) => a + b, 0) / allBytesMeasurements.length).toFixed(2)
+    )
+    console.log(
+      `Average percentage of memory used by SDK for ${sdkTask} over ${NUMBER_OF_RUNS} runs: ${sdkMemoryPercentage}%  for ${sdkMemoryBytes} bytes`
+    )
+    results.push({ sdkTask, sdkMemoryBytes, sdkMemoryPercentage })
+  }
+  return results
+}
+
+async function runTest(i, buttonName, bundleUrl, benchmarkUrl) {
+  const browser = await puppeteer.launch({
+    args: ['--no-sandbox', '--disable-setuid-sandbox'],
+  })
+  const page = await browser.newPage()
+  await page.goto(benchmarkUrl)
+
+  // Start the Chrome DevTools Protocol session and enable the heap profiler
+  const client = await page.target().createCDPSession()
+  await client.send('HeapProfiler.enable')
+
+  // Select the button to trigger the sdk task
+  await page.waitForSelector('button')
+  const button = (await page.$$('button'))[i]
+
+  await client.send('HeapProfiler.collectGarbage')
+
+  // Start the heap profiler sampling
+  await client.send('HeapProfiler.startSampling', {
+    samplingInterval: 50,
+  })
+
+  console.log(`Running test for: ${buttonName}`)
+  await button.click()
+  await timeout(TASK_DURATION)
+  const { profile } = await client.send('HeapProfiler.stopSampling')
+  const measurementsPercentage = []
+  const measurementsBytes = []
+  const sizeForNodeId = new Map()
+  for (const sample of profile.samples) {
+    sizeForNodeId.set(sample.nodeId, (sizeForNodeId.get(sample.nodeId) || 0) + sample.size)
+    let totalSize = 0
+    let sdkConsumption = 0
+    for (const node of iterNodes(profile.head)) {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ Maybe we can find a more suitable name for the `iterNodes()` function? Something like `children(profile.head)` although it's not only children.

Reply:
Good idea ! I think it could be more clear indeed 👍